docs/guides/enforce-active-organization.mdx

```diff
@@ -0,0 +1,208 @@
+    // Include their current route as a redirect_url, which will be handled on the route
+    if (auth().userId && !auth().orgId && req.nextUrl.pathname !== '/organization-selection') {
+      const redirectUrl = encodeURIComponent(req.nextUrl.href)
+      const orgListUrl = new URL(`/organization-selection?redirect_url=${redirectUrl}`, req.url)
+      return NextResponse.redirect(orgListUrl)
+    }
+
+    // If the user isn't signed in and the route is private, redirect to sign-in
+    if (!auth().userId && !isPublicRoute(req))
+      return auth().redirectToSignIn({ returnBackUrl: req.url })
+
+    // If the user is logged in and the route is protected, let them view.
+    if (auth().userId && !isPublicRoute(req)) return NextResponse.next()
+  })
+
+  export const config = {
+    matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+  }
+  ```
+
+  ### Updating `/organization-selection`
+
+  With Middleware adding the `redirect_url` search parameter to the URL, the route can then consume that and use it once the user has selected or created an organization.
+
+  First, modify the component parameters to ready the search parameters. Additionally indicate that the parameter expected is `redirect_url` with a type of string.
+
+  Second, once `orgId` is truthy check `redirect_url` was passed as a search parameter. If a `redirect_url` was passed then use that value, after URL decoding it, for the redirect. If there is no `redirect_url` then use `/dashboard` as a fallback.
+
+  ```tsx {{ filename: "/app/organization-selection", mark: [[6, 10], [13,21]]}}
+
+  import { OrganizationList } from "@clerk/nextjs";
+  import { auth } from "@clerk/nextjs/server";
+  import { redirect } from "next/navigation";
+
+  export default async function OrganizationSelectionPage({
+    searchParams
+  }: {
+    searchParams: { redirect_url: string }
+  }) {
+    const { orgId } = auth()
+
+    if (orgId) {
+      // If the orgId is truthy, then redirect the user to the redirect_url if present
+      if (searchParams.redirect_url) {
+        redirect(decodeURIComponent(searchParams.redirect_url))
+      }
+
+      // If there is no redirect_url, redirect to /dashboard
+      redirect("/dashboard")
+    }
+
+    return (
+      <>
+        <h1>Select an organization</h1>
+        <OrganizationList hidePersonal />
+      </>
+    )
+
+  }
+  ```
+</Steps>
+
+## Wrap up
+
+This will now reliably detect any user without an active organization (`orgId`), whether that user is someone who has just signed up or someone who organization was deleted or who left an organization. Any user without an active organization will be redirected and forced to create or select and organization.
+
+The route can modified to fit the needs of your application. You could configure it so the user can join an organization based on [email domain](/docs/organizations/verified-domains), or require a user to subscribe to a plan in your application before creating an organization.
```

Comment:
```suggestion
The route can modified to fit the needs of your application. You could configure it so that the user can join an organization based on an [email domain](/docs/organizations/verified-domains), or require a user to subscribe to a plan in your application before creating an organization.
```